purchase.adoc

```diff
@@ -388,288 +517,533 @@ Fields named "*UUID1" are the standard UUID Type 1 (8-4-4-4-12) representation o
+----
 
-*Sum of discounted rows* = 20 x 2 + 10 x 10 - 15 = 125.
+*Note:* Currently, we only support one discount for the whole, so the `discounts` list will only contain one item with `quantity` set to 1. This could potentially change in the future.
 
-*Cart discount* = 125 x 0.2 = 25
-
-So the total discount on this purchase is: 15 + 25 = 40. And the total sum to pay is: 140 - 40 = 100.
-
-A discount can never be greater than the actual amount of the row or cart price.
-
-### How to load all purchases for a user
+See further below for a full response example containing discounts.
+--
 
-You should never retrieve all the purchases in one request, since that can potentially put too much load on the server causing the request to be rejected or time out.
+*Discount calculation example*
 
-Instead, you should request the purchases splitted up into "pages", using the `limit` and `lastPurchaseHash` parameters, with `limit` set to maximum 1000.
+Let's observe an example purchase containing both discounts on rows as well as on the purchase itself.
 
-The `limit` parameter will set the page size, and the `lastPurchaseHash` parameter sets the starting point from where to retrieve purchases. Every response from the purchase service will contain a `lastPurchaseHash` field, which can be added to the new request to retrieve the next "page". A prepared URL is also provided in attribute `linkUrls`, if more purchases are to be fetched.
+[cols="3%,20%,20%,20%,27%"]
+|===
+|Nr|Product|Unit price|Amount|Discount
+|1|Foo|20|2|5
+|2|Bar|10|10|10%
+4+|Purchase discount| 20%
+|===
 
-In order to load all purchases, begin with an initial request without the `lastPurchaseHash` parameter being set, this will retrieve the first _n_ purchases (where _n_ is the page size defined by the `limit` parameter):
+1. To calculate the total discount we start by calculating the discount per row:
++
+--
+1 st row contains 2 items and a fixed discount amount of 5.
 
-```
-GET /purchases/v2?limit=50
-```
+2nd row contains 10 items and a percentage discount of 10%.
 
-The result will contain the first _n_ number of purchases.
-```json
-{
-  "purchases": [
-    ....
-  ],
-  "firstPurchaseHash": "14233908814696HbDrnUNRji5iniGikNLiQ",
-  "lastPurchaseHash": "1423390928355zj9yI1wyTvqP46AG8NEaYg",
-  "linkUrls": [
-        "<https://purchase.izettle.com/purchases/v2?limit=10&descending=true&lastPurchaseHash=1423390928355zj9yI1wyTvqP46AG8NEaYg>; rel=\"next\""
-  ]
-}
-```
+*Row discounts* = 5 + 10 x 10 x 0.1 = 15.
+--
+2. Then we calculate the purchase discount, based on the sum of the previously discounted rows:
++
+--
+*Sum of discounted rows* = 20 x 2 + 10 x 10 - 15 = 125
 
-To retrieve the following pages, use the value of `linkUrls` in the previous response to load the next page:
-```
-GET https://purchase.izettle.com/purchases/v2?limit=10&descending=true&lastPurchaseHash=1423390928355zj9yI1wyTvqP46AG8NEaYg
-```
+*Purchase discount* = 125 x 0.2 = 25
```

Comment:
Is it appropriate to use the unicode multiplication sign '×' instead of the letter 'x' throughout? I'm not sure myself, but like the way this is improving and thought that such a thing might add to the polish...